src/plugins/ucx/ucx_backend.cpp

```diff
@@ -352,7 +181,7 @@ void nixlUcxEngine::progressFunc()
 {
     using namespace nixlTime;
 
-    vramApplyCtx();
+    cudaMemCtx->set();
```

Comment:
Do we need to handle the case when the context has been already set outside the function?
If so, then push/pop maybe used to return to the original state after returning from the function, or maybe push only if there is no context bound to the thread.

Reply:
Please ignore this comment. I took a closer look at the function `progressFunc` and how it is used in the class.